src/libutil/posix-source-accessor.hh

```diff
@@ -28,7 +38,22 @@ struct PosixSourceAccessor : virtual SourceAccessor
 
     std::string readLink(const CanonPath & path) override;
 
-    std::optional<CanonPath> getPhysicalPath(const CanonPath & path) override;
+    std::optional<std::filesystem::path> getPhysicalPath(const CanonPath & path) override;
+
+    /**
+     * Create a `PosixSourceAccessor` and `CanonPath` corresponding to
+     * some native path.
+     *
+     * The `PosixSourceAccessor` is rooted as far up the tree as
+     * possible, (e.g. on Windows it could scoped to a drive like
+     * `C:\`). This allows more `..` parent accessing to work.
+     *
+     * See
+     * [`std::filesystem::path::root_path`](https://en.cppreference.com/w/cpp/filesystem/path/root_path)
+     * and
+     * [`std::filesystem::path::relative_path`](https://en.cppreference.com/w/cpp/filesystem/path/relative_path).
+     */
+    static std::pair<PosixSourceAccessor, CanonPath> openWithinRoot(const std::filesystem::path & path);
```

Comment:
The name `openWithinRoot` seems suboptimal, since nothing is being opened. Maybe `createAtRoot`?

Reply:
Yeah fair enough. `PosixSourceAccessor` doesn't (yet :)) keep a root directory file descriptor open :).